loris/webapp.py

```diff
@@ -621,19 +621,18 @@ def _make_image(self, image_request, src_fp, src_format):
             (str) the fp of the new image
         '''
         # figure out paths, make dirs
-        if self.enable_caching:
-            target_fp = self.img_cache.create_dir_and_return_file_path(image_request)
-        else:
-            # random str
-            n = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
-            target_fp = '%s.%s' % (path.join(self.tmp_dp, n), image_request.format)
+        n = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
+        temp_fp = '%s.%s' % (path.join(self.tmp_dp, n), image_request.format)
```

Comment:
For the deletion issues, you can pass delete=False, and then the file doesn't get deleted automatically - you have to do it yourself.

For the file being re-opened while the temp file is still open, the documentation does say that it works on Unix, which is all we support in Loris:
>  (it can be so used on Unix; it cannot on Windows NT or later)

Seems like the problems you ran into are fix-able, and this change would use the function provided by Python for creating a temp file, instead of rolling our own temp file names.

But, if @jpstroop wants to merge like this, I won't try to block the PR.